mm2src/coins_activation/src/tendermint_with_assets_activation.rs

```diff
@@ -217,6 +229,37 @@ impl From<TendermintInitError> for EnablePlatformCoinWithTokensError {
     }
 }
 
+async fn activate_with_walletconnect(
+    ctx: &MmArc,
+    session_topic: String,
+    chain_id: &str,
+    ticker: &str,
+) -> MmResult<(TendermintActivationPolicy, TendermintWalletConnectionType), TendermintInitError> {
+    let wc = WalletConnectCtx::from_ctx(ctx).expect("TODO: handle error when enable kdf initialization without key.");
+    let account = cosmos_get_accounts_impl(&wc, &session_topic, chain_id)
+        .await
+        .mm_err(|err| TendermintInitError {
+            ticker: ticker.to_string(),
+            kind: TendermintInitErrorKind::UnableToFetchChainAccount(err.to_string()),
+        })?;
+    let wallet_type = if wc.is_ledger_connection() {
+        TendermintWalletConnectionType::WcLedger(session_topic)
+    } else {
```

Comment:
we should check weather this session topic is a ledger connection and not the active session (and should remove the active session field anyway)

Reply:
done